docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+# Immutability in JavaScript
+
+According to (wikipédia)[https://en.wikipedia.org/wiki/Immutable_object] an *immutable object (unchangeable object) is an object whose state cannot be modified after it is created*. This rule is quite simple, if you want to modify some property of an object you have to do it on a copy. We will see a bit later what improvements and fancy features it unlocks for our developments.
+
+## EcmaScript
+
+EcmaScript provides utilities to keep our data immutable. APIs of arrays and objects have methods to create copies and even prevent instances to be updated. More recently EcmaScript brings us a brand new syntax that makes it easier to achieve it.
+
+### Object.assign
+
+We want to add a name property in our object.
+
+https://gist.github.com/frinyvonnick/a281f15288d3ab90ce5354f01df52ccf
+
+We can do it with `Object.assign` and a little trick. Basically it copies all properties of an object to another one, thus it mutates the target object. So we use a small trick, by passing an empty object at first parameter we create a new reference.
+
+https://gist.github.com/frinyvonnick/98878e45d6a8ccb4618238dbec175ae9
+
+So we now have a new object with our new name property and a commonNames property remaining unchanged. With this method you can create/overwrite multiple properties at the same time.
+
+### Array.concat
+
+Now let's do it with an array. We have an array and we want to add two new elements in an immutable way.
+
+https://gist.github.com/frinyvonnick/461d4dbad6bff965a60b734f86619faa
+
+`Array.concat` doesn't mutate our array in contrary of `Array.push` an return a new array without using trick like we did with `Object.assign`.
```

Comment:
`Array.concat` doesn't mutate our array in contrary of `Array.push`: it returns a new array without using a trick like we did with `Object.assign`